src/hooks/useSupplyWithdrawModal/Modal/Supply/form.tsx

```diff
@@ -0,0 +1,281 @@
+/** @jsxImportSource @emotion/react */
+import { Typography } from '@mui/material';
+import BigNumber from 'bignumber.js';
+import {
+  BorrowBalanceAccountHealth,
+  Delimiter,
+  FormikSubmitButton,
+  FormikTokenTextField,
+  IsolatedAssetWarning,
+  LabeledInlineContent,
+  LabeledInlineContentProps,
+  ValueUpdate,
+  toast,
+} from 'components';
+import { VError, formatVErrorToReadableString } from 'errors';
+import React, { useMemo } from 'react';
+import { useTranslation } from 'translation';
+import { Asset } from 'types';
+import {
+  calculateCollateralValue,
+  calculateDailyEarningsCents,
+  calculateYearlyEarningsForAssets,
+  convertTokensToWei,
+  formatTokensToReadableValue,
+  getBigNumber,
+} from 'utilities';
+
+import { SAFE_BORROW_LIMIT_PERCENTAGE } from 'constants/safeBorrowLimitPercentage';
+import { TOKENS } from 'constants/tokens';
+import { AmountForm, AmountFormProps, ErrorCode } from 'containers/AmountForm';
+
+import { useStyles } from '../styles';
+import TEST_IDS from './testIds';
+
+interface SupplyFormUiProps {
+  asset: Asset;
+  assets: Asset[];
+  tokenInfo: LabeledInlineContentProps[];
+  maxInput: BigNumber;
+  userTotalBorrowBalanceCents: BigNumber;
+  userTotalBorrowLimitCents: BigNumber;
+  inputLabel: string;
+  enabledButtonKey: string;
+  disabledButtonKey: string;
+  calculateNewBalance: (initial: BigNumber, amount: BigNumber) => BigNumber;
+  isTransactionLoading: boolean;
+  amountValue: string;
+}
+
+export const SupplyContent: React.FC<SupplyFormUiProps> = ({
+  asset,
+  tokenInfo,
+  userTotalBorrowBalanceCents,
+  userTotalBorrowLimitCents,
+  assets,
+  maxInput,
+  inputLabel,
+  enabledButtonKey,
+  disabledButtonKey,
+  calculateNewBalance,
+  isTransactionLoading,
+  amountValue,
+}) => {
+  const styles = useStyles();
+  const { t, Trans } = useTranslation();
+
+  const amount = new BigNumber(amountValue || 0);
+  const validAmount = amount && !amount.isZero() && !amount.isNaN();
+
+  const hypotheticalTokenSupplyBalance = amountValue
+    ? calculateNewBalance(asset.userSupplyBalanceTokens, amount)
+    : undefined;
+
+  const hypotheticalBorrowLimitCents = useMemo(() => {
+    const tokenPriceDollars = getBigNumber(asset?.tokenPriceDollars);
+    let updateBorrowLimitCents;
+
+    if (tokenPriceDollars && validAmount) {
```

Comment:
I know that's from the previous code, but I think we should remove `getBigNumber` here. Its usage is a bit confusing, since it will return a 0 BigNumber if the parameter its given is `undefined`. Let me know your opinion, but I think we should get rid of the `getBigNumber` util (it's only used in the supply/withdraw modals).

 ```suggestion
    let updateBorrowLimitCents;

    if (asset?.tokenPriceDollars && validAmount) {
```

Reply:
Agreed and removed, as it was it was just making it a bit harder to follow the logic.

Reply:
Thanks!